core/src/components/datetime/utils/format.ts

```diff
@@ -1,5 +1,28 @@
 import { DatetimeParts } from '../datetime-interface';
 
+const get12HourTime = (hour: number) => {
+  return hour % 12 || 12;
+}
+
+const getFormattedAMPM = (ampm?: string) => {
+  if (ampm === undefined) { return ''; }
+
+  return ampm.toUpperCase();
+}
+
+export const getFormattedTime = (refParts: DatetimeParts, use24Hour: boolean): string => {
+  if (refParts.hour === undefined || refParts.minute === undefined) { return 'Invalid Time'; }
```

Comment:
Should this also warn if the user tries to pass a logistically invalid time, like `25` for the hour?

Reply:
We could, but the `refParts` that gets passed in is created via users selecting options in datetime that we create so there's no user generated content there. I believe datetime will error out if you pass in an invalid ISO 8601 string into the `value` field for user generated content. I can add a check if you think it would be valuable though.

Reply:
Nah, if the only way to enter a custom time is through an ISO string, we should be good 👍 Just wanted to make sure.